Library/ViewModels/RewardPledgeViewModel.swift

```diff
@@ -224,20 +230,28 @@ RewardPledgeViewModelOutputs {
       .map { AppEnvironment.current.currentUser }
       .skipRepeats(==)
 
-    let shippingRules = projectAndReward
-      .switchMap { (project, reward) -> SignalProducer<[ShippingRule], NoError> in
+    let shippingRulesEvent = projectAndReward
+      .switchMap { (project, reward) -> SignalProducer<Event<[ShippingRule], ErrorEnvelope>, NoError> in
         guard reward != Reward.noReward else {
-          return SignalProducer<[ShippingRule], NoError>(value: [])
+          return SignalProducer(value: .value([]))
         }
 
         return AppEnvironment.current.apiService.fetchRewardShippingRules(
           projectId: project.id, rewardId: reward.id
           )
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
           .map(ShippingRulesEnvelope.lens.shippingRules.view)
-          .demoteErrors()
+          .retry(upTo: 3)
+          .materialize()
     }
 
+    self.shippingIsLoading = Signal.merge(
+      projectAndReward.map { _, reward in reward != Reward.noReward },
+      shippingRulesEvent.filter { $0.isTerminating }.mapConst(false)
+    )
```

Comment:
hey @mbrandonw i just modified your suggestion a bit to map to the `noReward` condition since we don't want it animating if the request is never made.